staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go

```diff
@@ -225,12 +225,12 @@ func (f CodecFactory) CodecForVersions(encoder runtime.Encoder, decoder runtime.
 
 // DecoderToVersion returns a decoder that targets the provided group version.
 func (f CodecFactory) DecoderToVersion(decoder runtime.Decoder, gv runtime.GroupVersioner) runtime.Decoder {
-	return f.CodecForVersions(nil, decoder, nil, gv)
+	return f.CodecForVersions(nil, decoder, runtime.DisabledGroupVersioner, gv)
 }
```

Comment:
if the encode is nil, the default is `runtime.DisabledGroupVersioner`. so should we given the value?

Reply:
These `nil` fallbacks were apparently for backwards-compatibility:

https://github.com/kubernetes/kubernetes/blob/25e786bf8ece51bfa1797b85dc0eaa19606c3466/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go#L216-L222

It looked more correct to do this, but I don't feel strongly about it.